File: C++/Easy/Make the String Great.cpp
//O(N) time, O(N) space
class Solution {
public:
    string makeGood(string s) {
        string stack;
        for (int i = 0; i < s.size(); i++) {
            stack.push_back(s[i]);
            if (stack.size() >= 2) {
                if (abs(stack[stack.size() - 2] - stack.back()) == 32) {
                    stack.pop_back();
                    stack.pop_back();
                }
            }
        }
        return stack;
    }
};

File: Python/Make the String Great.py
class Solution:
    def makeGood(self, s: str) -> str:
        list = []
        for char in s:
            list.append(char)
            if len(list) >= 2:
                if abs(ord(list[-1]) - ord(list[-2])) == 32:
                    list.pop()
                    list.pop()
        return "".join(list)
                